clients/ts/signalr/src/HandshakeProtocol.ts

```diff
@@ -55,7 +55,11 @@ export class HandshakeProtocol {
 
         // At this point we should have just the single handshake message
         const messages = TextMessageFormat.parse(messageData);
-        responseMessage = JSON.parse(messages[0]);
+        const response = JSON.parse(messages[0]);
+        if (response.type) {
+            throw new Error("Handshake response should not have a 'type' value.");
```

Comment:
What is this?

Reply:
This is checking if we receive a message that isn't a handshake response, we do the same thing in the .NET Client

Reply:
Agreed. Perhaps `Expected a handshake response from the server`.

Reply:
Shall I change the .NET error too then?